4/LR4_Task1.py

```diff
@@ -0,0 +1,254 @@
+        Открывает багажник автомобиля.
+
+        :return: Сообщение о том, что багажник открыт.
+        """
+        return "Багажник открыт"
+
+    def accelerate(self, increment: float) -> None:
+        """
+        Перегруженный метод для увеличения скорости автомобиля.
+        У легковых автомобилей ускорение ограничено 120 км/ч.
+
+        :param increment: Величина увеличения скорости.
+        """
+        self.speed = min(120, self.speed + increment)
+
+
+class Truck(Transport):
+    """
+    Дочерний класс для грузовых автомобилей.
+
+    Атрибуты:
+        _load_capacity (float): Грузоподъемность в тоннах.
+    """
+
+    def __init__(self, brand: str, model: str, year: int, load_capacity: float, speed: float = 0.0):
+        """
+        Конструктор класса Truck.
+
+        :param brand: Марка грузовика.
+        :param model: Модель грузовика.
+        :param year: Год выпуска.
+        :param load_capacity: Грузоподъемность в тоннах.
+        :param speed: Текущая скорость (по умолчанию 0.0).
+        """
+        super().__init__(brand, model, year, speed)
+        self._load_capacity = load_capacity
+
+    @property
+    def load_capacity(self) -> float:
+        """Возвращает грузоподъемность грузовика."""
+        return self._load_capacity
+
+    @load_capacity.setter
+    def load_capacity(self, value: float) -> None:
+        """
+        Устанавливает грузоподъемность грузовика.
+        Грузоподъемность должна быть положительным числом.
+        """
+        if value <= 0:
+            raise ValueError("Грузоподъемность должна быть положительным числом.")
+        self._load_capacity = value
+
+    def __str__(self) -> str:
+        """
+        Возвращает строковое представление объекта.
+
+        :return: Строка с описанием грузовика.
+        """
+        return f"{self.brand} {self.model} ({self.year}), грузоподъемность: {self.load_capacity} тонн, текущая скорость: {self.speed} км/ч"
+
+    def __repr__(self) -> str:
+        """
+        Возвращает формальное строковое представление объекта.
+
+        :return: Формальная строка с описанием грузовика.
+        """
+        return f"Truck(brand={self.brand}, model={self.model}, year={self.year}, load_capacity={self.load_capacity}, speed={self.speed})"
+
+    def load_cargo(self, weight: float) -> str:
+        """
+        Загружает груз в грузовик.
+
+        :param weight: Вес груза в тоннах.
+        :return: Сообщение о результате загрузки.
+        """
+        if weight <= self.load_capacity:
+            return f"Груз весом {weight} тонн успешно загружен."
+        else:
+            return f"Груз весом {weight} тонн превышает грузоподъемность."
+
+    def brake(self, decrement: float) -> None:
+        """
+        Перегруженный метод для уменьшения скорости грузовика.
+        У грузовиков торможение происходит медленнее.
+
+        :param decrement: Величина уменьшения скорости.
+        """
+        self.speed = max(0, self.speed - decrement * 0.5)
```

Comment:
В данном случае возможна ошибка, когда мы получим отрицательное число. Так что лучше написать вот так: 
```suggestion
        self.speed = max(0, self.speed - min(decrement * 0.5, self.speed))
```